File: output.pl
#!/usr/local/bin/perl

package nanobot::output;

my $RED		= "\033[31m";
my $GREEN	= "\033[32m";
my $YELLOW	= "\033[33m";
my $BLUE	= "\033[34m";
my $END		= "\033[0m";

sub new {
	my $class = shift;
	my $self = {
		_status		=> shift
	};

	bless $self;
	return $self;
}

sub std {
	my( $self, $message ) = @_;

	if( $self->{_status}->getoutput == 1) {
		print STDOUT "${message}";
	}
}

sub info {
	my( $self, $message ) = @_;

	if( $self->{_status}->getoutput == 1) {
		if( $self->{_status}->getcolour == 1) {
			print STDOUT "$YELLOW$message$END";
		} else {
			print STDOUT "${message}";
		}
	}
}

sub good {
	my( $self, $message ) = @_;

	if( $self->{_status}->getoutput == 1) {
		if( $self->{_status}->getcolour == 1) {
			print STDOUT "$GREEN$message$END";
		} else {
			print STDOUT "${message}";
		}
	}
}

sub bad {
	my( $self, $message ) = @_;

	if( $self->{_status}->getoutput == 1) {
		if( $self->{_status}->getcolour == 1) {
			print STDOUT "$RED$message$END";
		} else {
			print STDOUT "${message}";
		}
	}
}

sub special {
	my( $self, $message ) = @_;

	if( $self->{_status}->getoutput == 1) {
		if( $self->{_status}->getcolour == 1) {
			print STDOUT "$BLUE$message$END";
		} else {
			print STDOUT "${message}";
		}
	}
}

sub debug {
	my( $self, $message ) = @_;

	if( $self->{_status}->getdebug >= 1) {
		print STDOUT "${message}";
	}
}

sub extradebug {
	my( $self, $message ) = @_;

	if( $self->{_status}->getdebug == 2) {
		print STDOUT "${message}";
	}
}
1;

File: output.rb
#!/usr/bin/ruby

# Class used to send output to terminal
class Output
	def initialize( status )
		@status		= status

		@RED		= "\033[31m";
		@GREEN		= "\033[32m";
		@YELLOW		= "\033[33m";
		@BLUE		= "\033[34m";
		@END		= "\033[0m";
	end

	def std( string )
		if( @status.output == 1 )
			$stdout.print(string)
		end
	end

	def info( string )
		if( @status.output == 1 )
			if( @status.colour == 1 )
				$stdout.print(@YELLOW + string + @END)
			else
				$stdout.print(string)
			end
		end
	end

	def special( string )
		if( @status.output == 1 )
			if( @status.colour == 1 )
				$stdout.print(@BLUE + string + @END)
			else
				$stdout.print(string)
			end
		end
	end

	def good( string )
		if( @status.output == 1 )
			if( @status.colour == 1 )
				$stdout.print(@GREEN + string + @END)
			else
				$stdout.print(string)
			end
		end
	end

	def bad( string )
		if( @status.output == 1 )
			if( @status.colour == 1 )
				$stdout.print(@RED + string + @END)
			else
				$stdout.print(string)
			end
		end
	end

	def debug( string )
		if( @status.debug >= 1 )
			$stdout.print(string)
		end
	end

	def debug_extra( string )
		if( @status.debug == 2 )
			$stdout.print(string)
		end
	end

end
